### PR TITLE
Add test coverage reporting

### PR DESCRIPTION
**Pull request by johncornish**

### CODE REVIEW

**Comment by johncornish** [comment]
Also includes code coverage hackery

**Review by mmichael — COMMENTED** [review]
Did you want to check in all of the coverage/* files ?

File: Makefile [hunk 1]
```diff
@@ -157,6 +168,23 @@ deploy-test: token-check k8s-env clean-deployment deploy-targets proxy-test
 
 #Testing code, configuration and deployment changes
 integration-test: token-check k8s-env clean-deployment deploy-targets delete-images push-images proxy-test
+	echo "Integration test finished. To generate code coverage reports, run 'make coverage-test'"
```
Comment (mmichael) [inline Makefile:171]: I'd suggest removing this comment, it will become a bit redundant after a while

Comment (johncornish) [reply Makefile:171]: Will do

**Pull request merged**